src/scripts/main.js

```diff
@@ -1 +1,89 @@
 'use strict';
+
+function createMessage(text, className = `success`) {
+  const div = document.createElement('div');
+
+  div.classList = className;
+  div.setAttribute('data-qa', 'notification');
+  div.textContent = text;
+  document.body.append(div);
+}
+
+function addClickHandlers(mouseClick) {
+  document.addEventListener('click', mouseClick);
+  document.addEventListener('contextmenu', mouseClick);
+}
+
+function removeClickHandlers(mouseClick) {
+  document.removeEventListener('click', mouseClick);
+  document.removeEventListener('contextmenu', mouseClick);
+}
+
+const firstPromise = new Promise((resolve, reject) => {
+  const leftClick = (e) => {
+    if (e.button === 0) {
+      resolve(`First promise was resolved`);
+      document.removeEventListener('click', leftClick);
+    }
+  };
+
+  document.addEventListener('click', leftClick);
+
+  setTimeout(() => {
+    reject(new Error(`First promise was rejected`));
+    document.removeEventListener('click', leftClick);
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  const mouseClick = (e) => {
+    if (e.button === 2) {
+      e.preventDefault();
+    }
+
+    if (e.button === 0 || e.button === 2) {
+      resolve(`Second promise was resolved`);
+    }
+
+    removeClickHandlers(mouseClick);
```

Comment:
The `removeClickHandlers` function is called within the `mouseClick` function for `secondPromise`, which might cause issues if `thirdPromise` is also active, as it will remove the handlers for both promises. Consider using separate handlers or a different mechanism to manage event listeners for each promise.